org.testeditor.web.dropwizard/src/main/java/org/testeditor/web/dropwizard/DropwizardApplication.xtend

```diff
@@ -93,9 +98,12 @@ abstract class DropwizardApplication<T extends Configuration> extends Applicatio
 	}
 
 	protected def void configureAuthFilter(T configuration, Environment environment) {
-		val filter = authFilterBuilder.buildAuthFilter
+		val List<AuthFilter> authFilters = #[authFilterBuilder.buildAuthFilter, apiTokenAuthFilterBuilder.buildAuthFilter] // ChainedAuthFilter expects this without explicit AuthFilter<String,User>
+		// see http://www.dropwizard.io/1.2.0/docs/manual/auth.html#chained-factories
+		// and http://dinomite.net/blog/2016/05/18/optional-authentication-with-dropwizard/
+		val chainedAuthFilter = new DefaultAuthDynamicFeature(new ChainedAuthFilter(authFilters))
```

Comment:
That comments would be better of by residing within `DefaultAuthDynamicFeature`. We could also inject the `@Inject JWTAuthFilter.Builder authFilterBuilder` and `@Inject ApiTokenAuthFilter.Builder apiTokenAuthFilterBuilder` into that class and just use it here.

Reply:
The DefaultAuthDynamicFeature filters the list of authenticators. Putting the Auth filters into that class would make it less general. The composition itself could be put into a separate method or alltogether into a different class but I would not change the dynamic feature itself, making it less composable.